rotkehlchen/globaldb/handler.py

```diff
@@ -230,16 +231,31 @@ def retrieve_assets(filter_query: 'AssetsFilterQuery') -> Tuple[List[Dict[str, A
         - DeserializationError
         """
         assets_info = []
-        query, bindings = filter_query.prepare()
+        underlying_tokens: Dict[str, List[Dict[str, str]]] = defaultdict(list)
+        prepared_filter_query, bindings = filter_query.prepare()
         parent_query = """
         SELECT A.identifier AS identifier, A.type, B.address, B.decimals, A.name, C.symbol, C.started, C.forked, C.swapped_for, C.coingecko, C.cryptocompare, B.protocol, B.chain, B.token_kind, D.notes, D.type AS custom_asset_type
         FROM assets as A
         LEFT JOIN common_asset_details AS C ON C.identifier = A.identifier
         LEFT JOIN evm_tokens as B ON B.identifier = A.identifier
         LEFT JOIN custom_assets as D ON D.identifier = A.identifier
         """  # noqa: E501
-        query = f'SELECT * FROM ({parent_query}) {query}'
+        query = f'SELECT * FROM ({parent_query}) {prepared_filter_query}'
+        underlying_tokens_query = """
+        SELECT A.identifier AS identifier, A.type, B.address, B.decimals, A.name, C.symbol, C.started, C.forked, C.swapped_for,
+        C.coingecko, C.cryptocompare, B.protocol, B.chain, B.token_kind, D.notes, D.type AS custom_asset_type,
+        E.parent_token_entry as underlying_token_parent, E.weight AS underlying_token_weight
+        FROM underlying_tokens_list AS E
+        JOIN assets as A ON A.identifier = E.parent_token_entry
+        JOIN evm_tokens as B ON A.identifier=B.identifier
+        JOIN common_asset_details AS C ON C.identifier = A.identifier
+        LEFT JOIN custom_assets as D ON D.identifier = A.identifier
```

Comment:
why are you joining the custom assets table here? You are searching only for evm tokens

Reply:
If I don't join other tables, the query will fail because it requires those columns.

Reply:
Ah is the type column, okey didn't see it